extensions/sql/asset/sql_asset_index_loader_proposal.md

```diff
@@ -0,0 +1,186 @@
+
+Given example Asset data
+
+<table>
+  <thead>
+    <tr>
+      <td><b>id</b></td>
+      <td><b>asset:prop:name</b></td>
+      <td><b>asset:prop:description</b></td>
+      <td><b>asset:prop:version</b></td>
+      <td><b>asset:prop:contenttype</b></td>
+      <td><b>data-address</b></td>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>784cdfde-a50c-11ec-b909-0242ac120002</td>
+      <td>Max Power</td>
+      <td>Homer to the Max</td>
+      <td><i>NULL</i></td>
+      <td><i>NULL</i></td>
+      <td>
+        <code>
+          {
+            "uri": "https://projects.eclipse.org/proposals/eclipse-dataspace-connector"
+          }
+        </code>
+      </td>
+    </tr>
+    <tr>
+      <td>80b69de0-a50c-11ec-b909-0242ac120002</td>
+      <td>Seneca</td>
+      <td>Seneca the Younger</td>
+      <td><i>v2</i></td>
+      <td><i>application/pdf</i></td>
+      <td>
+        <code>
+          {
+            "uri": "https://archive.org/download/adlucilium02sene/adlucilium02sene.pdf"
+          }
+        </code>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+would translate into following DML:
+
+```sql
+BEGIN;
+
+INSERT INTO edc_asset (asset_id) VALUES ("784cdfde-a50c-11ec-b909-0242ac120002");
+INSERT INTO edc_asset_property (asset_id, property_name, property_string_value) VALUES ("784cdfde-a50c-11ec-b909-0242ac120002", "asset:prop:name", "Max Power");
+INSERT INTO edc_asset_property (asset_id, property_name, property_string_value) VALUES ("784cdfde-a50c-11ec-b909-0242ac120002", "asset:prop:description", "Homer to the Max");
+INSERT INTO edc_asset_dataaddress (asset_id, properties) VALUES ("784cdfde-a50c-11ec-b909-0242ac120002", "{ \"uri\": \"https://projects.eclipse.org/proposals/eclipse-dataspace-connector\" }")
+
+INSERT INTO edc_asset (asset_id) VALUES ("80b69de0-a50c-11ec-b909-0242ac120002");
+INSERT INTO edc_asset_property (asset_id, property_name, property_string_value) VALUES ("80b69de0-a50c-11ec-b909-0242ac120002", "asset:prop:name", "Seneca");
+INSERT INTO edc_asset_property (asset_id, property_name, property_string_value) VALUES ("80b69de0-a50c-11ec-b909-0242ac120002", "asset:prop:description", "Seneca the Younger");
+INSERT INTO edc_asset_property (asset_id, property_name, property_string_value) VALUES ("80b69de0-a50c-11ec-b909-0242ac120002", "asset:prop:version", "v2");
+INSERT INTO edc_asset_property (asset_id, property_name, property_string_value) VALUES ("80b69de0-a50c-11ec-b909-0242ac120002", "asset:prop:contenttype", "application/pdf");
+INSERT INTO edc_asset_dataaddress (asset_id, properties) VALUES ("80b69de0-a50c-11ec-b909-0242ac120002", "{ \"uri\": \"https://archive.org/download/adlucilium02sene/adlucilium02sene.pdf\" }")
+
+
+COMMIT;
+```
+
+#### AssetLoader: `Asset deleteById(String assetId)`
+Soft deletion has been postponed. For noe, deletion will work as follows.
+
+```sql
+DELETE FROM edc_asset WHERE asset_id = "784cdfde-a50c-11ec-b909-0242ac120002";
+```
+
+
```

Comment:
Missing the `findById` case. Note: don't forget to join with data address and properties.